shortfin/python/shortfin_apps/llm/components/invocation.py

```diff
@@ -326,37 +326,33 @@ def get_args_data(
                 - A list of start positions.
                 - A list of sequence block IDs.
         """
+        task_input = self.task_input
         token_data = [
-            input_tokens
-            for req in exec_requests
-            for input_tokens in (req.input_token_ids[-1:])
+            padded_tokens
+            for tokens in task_input.input_tokens
+            for padded_tokens in (tokens[-1:])
         ]
-        seq_lens_data = [req.start_position + 1 for req in exec_requests]
-        start_positions_data = [req.start_position for req in exec_requests]
 
         seq_block_ids_data = []
-        for req in exec_requests:
-            block_ids = req.cache_page_indices(block_count)
-            # Pad the block IDs to match the block count.
-            padded = _pad_list(
-                block_ids,
-                target_length=block_count,
+        for pages in task_input.page_ids:
+            block_ids = _pad_list(
+                pages,
+                target_length=task_input.block_count,
             )
-            # Extend the sequence block IDs data with padded values.
-            seq_block_ids_data.extend(padded)
+            seq_block_ids_data.extend(block_ids)
 
         return (
             token_data,
-            seq_lens_data,
-            start_positions_data,
+            [position + 1 for position in task_input.start_positions],
```

Comment:
Don't manipulate types on return. The result should be independently defined. Return is just for returning.